packages-common/pluggable-widgets-tools/configs/webpack.config.dev.js

```diff
@@ -10,17 +12,53 @@ const name = widgetName.toLowerCase();
 const mxHost = variables.package.config.mendixHost || "http://localhost:8080";
 const developmentPort = variables.package.config.developmentPort || "3000";
 
+const isHotRefresh = process.argv.indexOf("--hot") !== -1;
+
+const tsLoaders = [
+    {
+        loader: "babel-loader",
+        options: {
+            cacheDirectory: true,
+            plugins: [require("react-refresh/babel")]
+        }
+    },
+    {
+        loader: "ts-loader"
+    }
+];
+
+if (!isHotRefresh) {
+    tsLoaders.shift();
+}
+
 const devConfig = {
     mode: "development",
     devtool: "source-map",
     devServer: {
+        contentBase: join(process.cwd(), "dist/tmp"),
+        compress: true,
         port: developmentPort,
         overlay: {
             errors: true
         },
+        before(app) {
+            app.get("/mxclientsystem/mxui/mxui.js", async (req, res) => {
+                const request = await fetch(`${mxHost}/mxclientsystem/mxui/mxui.js`);
+                const mxui = await request.text();
+                const { injectIntoGlobalHook } = require("react-refresh/runtime");
```

Comment:
It would be safer to load this code as text with `await readFile(require.resolve("react-refresh/runtime"))`. Because now you effectively evaluate it